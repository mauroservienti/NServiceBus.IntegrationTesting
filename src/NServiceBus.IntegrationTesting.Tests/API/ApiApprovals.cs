using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;

namespace NServiceBus.IntegrationTesting.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(IntegrationScenarioContext).Assembly.GeneratePublicApi();
            Approvals.Verify(publicApi);
        }
    }
}