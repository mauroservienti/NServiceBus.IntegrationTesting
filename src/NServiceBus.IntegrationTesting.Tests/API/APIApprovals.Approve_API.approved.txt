[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "https://github.com/mauroservienti/NServiceBus.IntegrationTesting.git")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("NServiceBus.IntegrationTesting.Tests")]
[assembly: System.Runtime.Versioning.TargetFramework(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace NServiceBus.IntegrationTesting
{
    public class EmptyTestCompletionHandler : NServiceBus.IntegrationTesting.IHandleTestCompletion
    {
        public EmptyTestCompletionHandler() { }
        public System.Threading.Tasks.Task OnTestCompleted(NServiceBus.AcceptanceTesting.Support.RunSummary summary) { }
    }
    public static class EndpointConfigurationExtensions
    {
        public static void RegisterRequiredPipelineBehaviors(this NServiceBus.EndpointConfiguration builder, string endpointName, NServiceBus.IntegrationTesting.IntegrationScenarioContext integrationScenarioContext) { }
        public static void RegisterScenarioContext(this NServiceBus.EndpointConfiguration builder, NServiceBus.AcceptanceTesting.ScenarioContext scenarioContext) { }
    }
    public abstract class EndpointTemplate : NServiceBus.AcceptanceTesting.Support.IEndpointSetupTemplate
    {
        protected EndpointTemplate() { }
        public System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> GetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
        protected abstract System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization);
    }
    public class EndpointTemplate<T> : NServiceBus.IntegrationTesting.EndpointTemplate
        where T : NServiceBus.EndpointConfiguration, new ()
    {
        public EndpointTemplate() { }
        protected override System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
    }
    public class EndpointTemplate<T, C> : NServiceBus.IntegrationTesting.EndpointTemplate<T>
        where T : NServiceBus.EndpointConfiguration, new ()
        where C : NServiceBus.IntegrationTesting.IHandleTestCompletion, new ()
    {
        public EndpointTemplate() { }
        protected override System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
    }
    public class HandlerInvocation : NServiceBus.IntegrationTesting.Invocation
    {
        public HandlerInvocation() { }
        public System.Type HandlerType { get; }
    }
    public interface IHandleTestCompletion
    {
        System.Threading.Tasks.Task OnTestCompleted(NServiceBus.AcceptanceTesting.Support.RunSummary summary);
    }
    public class IntegrationScenarioContext : NServiceBus.AcceptanceTesting.ScenarioContext
    {
        public IntegrationScenarioContext() { }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.HandlerInvocation> InvokedHandlers { get; }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.SagaInvocation> InvokedSagas { get; }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.OutgoingMessageOperation> OutgoingMessageOperations { get; }
        public bool HandlerWasInvoked<THandler>() { }
        public bool HasFailedMessages() { }
        public bool HasHandlingErrors() { }
        public bool MessageWasProcessed<TMessage>() { }
        public bool MessageWasProcessedByHandler<TMessage, THandler>() { }
        public bool MessageWasProcessedBySaga<TMessage, TSaga>() { }
        public void RegisterTimeoutRescheduleRule<TTimeout>(System.Func<object, NServiceBus.DelayedDelivery.DoNotDeliverBefore, NServiceBus.DelayedDelivery.DoNotDeliverBefore> rule) { }
        public bool SagaWasCompleted<TSaga>()
            where TSaga : NServiceBus.Saga { }
        public bool SagaWasInvoked<TSaga>()
            where TSaga : NServiceBus.Saga { }
    }
    public abstract class Invocation
    {
        protected Invocation() { }
        public string EndpointName { get; }
        public System.Exception HandlingError { get; }
        public object Message { get; }
        public System.Type MessageType { get; }
    }
    public abstract class OutgoingMessageOperation
    {
        protected OutgoingMessageOperation() { }
        public System.Collections.Generic.Dictionary<string, string> MessageHeaders { get; }
        public string MessageId { get; }
        public object MessageInstance { get; }
        public System.Type MessageType { get; }
        public System.Exception OperationError { get; }
        public string SenderEndpoint { get; }
    }
    public class PublishOperation : NServiceBus.IntegrationTesting.OutgoingMessageOperation
    {
        public PublishOperation() { }
    }
    public class ReplyOperation : NServiceBus.IntegrationTesting.OutgoingMessageOperation
    {
        public ReplyOperation() { }
    }
    public class RequestTimeoutOperation : NServiceBus.IntegrationTesting.SendOperation
    {
        public RequestTimeoutOperation() { }
        public string SagaId { get; }
        public string SagaTypeAssemblyQualifiedName { get; }
    }
    public class SagaInvocation : NServiceBus.IntegrationTesting.Invocation
    {
        public SagaInvocation() { }
        public bool IsCompleted { get; }
        public bool IsNew { get; }
        public bool NotFound { get; }
        public NServiceBus.IContainSagaData SagaData { get; }
        public System.Type SagaType { get; }
    }
    public class SendOperation : NServiceBus.IntegrationTesting.OutgoingMessageOperation
    {
        public SendOperation() { }
    }
}